src/rooms/rooms.service.ts

```diff
@@ -222,4 +240,84 @@ export class RoomsService {
         throw new Error(error);
       });
   }
+
+  async updateRoomQuestionStack(stackId: string, roomId: number) {
+    const roomResult: Room | undefined = await this.roomRepository.findOne({
+      where: { id: roomId },
+    });
+    if (!roomResult) return [null, 'Room Id is not corrected'];
+    const stackResult: QuestionStack | undefined =
+      await this.questionStackRepository.findOne({
+        where: {
+          id: stackId,
+          status: QuestionStackStatus.ACTIVE,
+        },
+      });
+    if (!stackResult) return [null, 'Stack Id is not corrected'];
+    const currentTime = new Date();
+    if (roomResult.openTime > currentTime)
+      return [null, 'Room is started, cant update any more'];
+    const currentStack = roomResult.questionStack;
+    currentStack.status = QuestionStackStatus.ACTIVE;
+    stackResult.status = QuestionStackStatus.USED;
+    roomResult.questionStack = stackResult;
+    try {
+      await this.dataSource.transaction(async (manager) => {
+        await manager.save(roomResult);
+        await manager.save(stackResult);
+        await manager.save(currentStack);
+      });
+    } catch (err) {
+      this.logger.error(err);
+      return [null, 'Errors when changing question stack'];
+    }
+    return [roomResult, null];
+  }
+
+  async addAllTeamsToRoom(roomTeam: CreateScoreTeamDto) {
+    if (!roomTeam?.roomCode) return [null, 'Enter room code and try again'];
+    const roomEntity: Room | undefined = await this.roomRepository.findOne({
+      where: {
+        code: roomTeam.roomCode,
+      },
+    });
+    if (!roomEntity)
+      return [null, `Not found room with code ${roomTeam.roomCode}`];
+    if (roomTeam.teamIds.length == 0) return ['Adding team successfully', null];
+    const teamList = await this.dataSource
+      .getRepository(Team)
+      .createQueryBuilder()
+      .select('*')
+      .where('id IN (:...ids)', { ids: roomTeam.teamIds })
+      .getMany();
+    for (const team of teamList) {
```

Comment:
Check if teamList is empty.

Reply:
I think not need this, but undefine is ok, let me do this